src/components/Registration/index.jsx

```diff
@@ -0,0 +1,97 @@
+import React, {useState, useEffect} from 'react';
+import './style.css';
+import userIcon from './img/user-solid.svg'
+
+const Registation = () => {
+
+    const [user, setUser] = useState({
+        email: "",
+        username: "",
+        password: "",
+        passwordConfirm: "",
+      });
+
+    const handleSubmit= (e) => {
+        e.preventDefault();
+        if (user.password === user.passwordConfirm && user.username !== '' && user.email !== '' && user.password !== '' && user.passwordConfirm !== '') {
+            console.log(user)
+          } else if (user.password !== user.passwordConfirm) {
+            alert('Your password and confirmation password do not match.')
+          } else {
+            alert('Please fill out all required fields.')
+          }
+    };
+
+    useEffect(() => {
+        if (user.email.indexOf("@") >= 0 && user.username.length === 0) {
+          setUser({ ...user, username: user.email.substring(0, user.email.indexOf("@")) });
+        }
+      }, [user.email]);
+
+     return (
+        <>
+        <div className="header">
+            <h1>Registration</h1>
+            <div className="userIcon">
+                <img src={userIcon} alt="user icon"/>
+            </div>
+        </div>
+
+        <div className="body">
+            <form className="form" onSubmit={handleSubmit}>
```

Comment:
dávat ten handleSubmit sem a ne dole do button onClick je lepší protože takhle jak to máš ty formulář reaguje i když zmáčkneš enter v políčku 👍